osier/technology.py

```diff
@@ -0,0 +1,258 @@
+    valid_quantity : :class:`unyt.unyt_quantity`
+        The validated quantity.
+    """
+    try:
+        exp_dim = _dim_opts[dimension]
+    except KeyError:
+        raise KeyError(f"Key <{dimension}> not accepted. Try: {_dim_opts}")
+
+    valid_quantity = None
+    if isinstance(value, unyt_quantity):
+        try:
+            valid_quantity = value.to(exp_dim)
+        except UnitConversionError:
+            raise TypeError(f"Cannot convert {value.units} to {exp_dim}")
+    elif isinstance(value, float):
+        valid_quantity = value * exp_dim
+    elif isinstance(value, int):
+        valid_quantity = value * exp_dim
+    elif isinstance(value, str):
+        try:
+            valid_quantity = float(value) * exp_dim
+        except ValueError:
+            try:
+                unyt_value = unyt_quantity.from_string(value)
+                assert unyt_value.units.same_dimensions_as(exp_dim)
+                valid_quantity = unyt_value
+            except UnitParseError:
+                raise UnitParseError(f"Could not interpret <{value}>.")
+            except AssertionError:
+                raise AssertionError(f"{value} lacks units of {dimension}.")
+    else:
+        raise ValueError(f"Value of type <{type(value)}> passed.")
+    return valid_quantity
+
+
+class Technology(object):
+    """
+    Parameters
+    ----------
+    technology_name : string
+        The name identifier of the technology.
+    technology_type : string
+        The string identifier for the type of technology.
+    capital_cost : float or :class:`unyt.array.unyt_quantity`
+        Specifies the capital cost. If float,
+        the default unit is $/MW.
+    om_cost_fixed : float or :class:`unyt.array.unyt_quantity`
+        Specifies the fixed operating costs.
+        If float, the default unit is $/MW.
+    om_cost_variable : float or :class:`unyt.array.unyt_quantity`
+        Specifies the variable operating costs.
+        If float, the default unit is $/MWh.
+    fuel_cost : float or :class:`unyt.array.unyt_quantity`
+        Specifies the fuel costs.
+        If float, the default unit is $/MWh.
+    capacity : float or :class:`unyt.array.unyt_quantity`
+        Specifies the technology capacity.
+        If float, the default unit is MW
+    default_power_units : str or :class:`unyt.unit_object.Unit`
+        An optional parameter, specifies the units
+        for power. Default is megawatts [MW].
+    default_time_units : str or :class:`unyt.unit_object.Unit`
+        An optional parameter, specifies the units
+        for time. Default is hours [hr].
+    default_energy_units : str or :class:`unyt.unit_object.Unit`
+        An optional parameter, specifies the units
+        for energy. Default is megawatt-hours [MWh]
+
+    Notes
+    -----
+    Cost values are listed in the docs as [$ / physical unit]. However,
+    :class:`osier` does not currently have a currency handler, therefore the
+    units are technically [1 / physical unit].
```

Comment:
Nothing needs to be renormalized, but if you wanted the data in euros you would have to convert the values yourself. It's not super relevant right now, but sub-classes of `Technology` will use data from NREL's ATBe (unless otherwise specified).